test/minter-suite-minters/MinterHolder/MinterHolderV2.test.ts

```diff
@@ -0,0 +1,476 @@
+import { expectRevert } from "@openzeppelin/test-helpers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import {
+  getAccounts,
+  assignDefaultConstants,
+  deployAndGet,
+  deployCoreWithMinterFilter,
+  compareBN,
+  safeAddProject,
+} from "../../util/common";
+
+import { MinterHolder_Common } from "./MinterHolder.common";
+import { smock, FakeContract } from "@defi-wonderland/smock";
+import { IDelegationRegistry } from "../../../scripts/contracts";
+
+// test the following V3 core contract derivatives:
+const coreContractsToTest = [
+  "GenArt721CoreV3", // flagship V3 core
+  "GenArt721CoreV3_Explorations", // V3 core explorations contract
+];
+
+/**
+ * These tests intended to ensure Filtered Minter integrates properly with V3
+ * core contract.
+ */
+for (const coreContractName of coreContractsToTest) {
+  describe(`MinterHolderV2_${coreContractName}`, async function () {
+    let fakeDelegationRegistry: FakeContract<IDelegationRegistry>;
```

Comment:
Similar to a previous comment, I think we should update this minter's architecture and tests to be in alignment with #370